numpy/random/mtrand/mtrand.pyx

```diff
@@ -620,13 +620,63 @@ def _rand_bool(low, high, size, rngstate):
         return array
 
 
-def _rand_int8(low, high, size, rngstate):
+def _rand_bool(low, high, size, rngstate):
     """
-    _rand_int8(low, high, size, rngstate)
+    _rand_bool(low, high, size, rngstate)
 
     See `_rand_int32` for documentation, only the return type changes.
 
     """
+    cdef npy_bool off, rng, buf
+    cdef ndarray array "arrayObject"
+    cdef npy_bool *array_data
+    cdef npy_bool *high_data
+    cdef npy_bool *low_data
+    cdef broadcast multi
+    cdef npy_intp i
+
+    cdef ndarray lo = <ndarray>low
+    cdef ndarray hi = <ndarray>high
+    cdef rk_state *state = <rk_state *>NpyCapsule_AsVoidPtr(rngstate)
+
+    if size is None:
+        multi = <broadcast>PyArray_MultiIterNew(2, <void *>lo, <void *>hi)
+        array = <ndarray>PyArray_SimpleNew(multi.nd, multi.dimensions, NPY_BOOL)
+        array_data = <npy_bool *>PyArray_DATA(array)
+        with nogil:
+            for i from 0 <= i < multi.size:
+                low_data = <npy_bool *>PyArray_MultiIter_DATA(multi, 0)
+                high_data = <npy_bool *>PyArray_MultiIter_DATA(multi, 1)
+                rng = <npy_bool>(high_data[0] - low_data[0])
+                off = <npy_bool>(low_data[0])
+
+                rk_random_bool(off, rng, 1, &buf, state)
+                array_data[i] = buf
+    else:
+        array = <ndarray>np.empty(size, np.bool_)
+        array_data = <npy_bool *>PyArray_DATA(array)
+        multi = <broadcast>PyArray_MultiIterNew(3, <void *>array, <void *>lo, <void *>hi)
```

Comment:
Sorry, never mind about this -- now that I'm at a proper keyboard and can play around, I see that all the random methods use this (IMO suboptimal) interpretation of `size=`, so `randint` should do the same for consistency.

(What I find particularly confusing is that it means that `size=None` and `size=()` are totally different. OTOH if size acted 'outside' of the core sampling operation, so that the parameters were broadcast against each other to define a single multidimensional "sample", and then size specified how many of this samples to take and then tile together, then the default would just be size=(). Also it would be easier to work with. But obviously the advantages here are too small to be worth breaking backcompat or consistency over. It might be worth going through and adding a new kwarg like `repeat=` that has the desired semantics without breaking compatibility, but it's obviously not super urgent and would be a different PR in any case.)